### PR TITLE
Improve error handling in runner.py `job.run(...)`

### LINKED ISSUE #8738: Exceptions in forward_model_runner will cause it to hang
Related to https://github.com/equinor/ert/issues/8729

Uncaught exceptions in the `forward_model_runner` will cause it to hang. This is most likely due to some of the reporters running in a separate thread that will never get the event signaling that they should exit.

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #8738 

**Approach**
This commit wraps the `job.run(...)` method in a try-except block making sure the `Exited` message is sent if the job_runner crashes. This fixes the bug where ert is hanging when unhandled exceptions are raised in the job runner.

The PR also contains a commit that refactors the mega-method `job.run(...)` into separate methods, making it more readable. 

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/_ert/forward_model_runner/job.py [hunk 1]
```diff
@@ -193,67 +202,118 @@ def ensure_file_handles_closed():
...
+            return exited_message
 
         # exit_code is 0
 
         if self.job_data.get("error_file") and os.path.exists(
             self.job_data["error_file"]
         ):
-            yield exited_message.with_error(
+            exited_message = Exited(self, exit_code)
+            return exited_message.with_error(
```
Comment (xjules) [inline src/_ert/forward_model_runner/job.py:242]: return?? previously we yielded ....

**Comment by xjules** [comment]
Lots of errors. Wondering if it is related that you have changed some `yield` statements with `return` :thinking:

**Comment by jonathan-eq** [comment]
> Lots of errors. Wondering if it is related that you have changed some `yield` statements with `return` 🤔

It should be fine. The methods returning values are just for creating the messages that are being yielded in the `_run` method.

File: src/_ert/forward_model_runner/job.py (hunk 1 above)
Comment (jonathan-eq) [reply src/_ert/forward_model_runner/job.py:241]: Yes, but this is a helper function for creating the message that is being yielded from the calling method.

**Review by JHolba — APPROVED** [review]

**Pull request merged**